File: iOS8_Adv_Training/PageView/PageView/DataViewController.m
//
//  DataViewController.m
//  PageView
//
//  Created by 新居雅行 on 2014/10/29.
//  Copyright (c) 2014年 msyk.net. All rights reserved.
//

#import "DataViewController.h"

@interface DataViewController ()

@end

@implementation DataViewController

- (void)viewDidLoad {
    NSLog( @"%s", __FUNCTION__ );
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)viewWillAppear:(BOOL)animated {
    NSLog( @"%s", __FUNCTION__ );
    [super viewWillAppear:animated];
    self.dataLabel.text = [self.dataObject description];

    NSString *fileName = [self.dataObject description];
    NSString *imagePath
    = [[NSBundle mainBundle] pathForResource: fileName
                                      ofType: @"jpg"];
    self.photoView.image
    = [UIImage imageWithContentsOfFile: imagePath];
    self.photoView.contentMode = UIViewContentModeScaleAspectFit;
}

@end

File: iOS8_Adv_Training/PageView_Swift/PageView_Swift/DataViewController.swift
//
//  DataViewController.swift
//  PageView_Swift
//
//  Created by 新居雅行 on 2015/01/29.
//  Copyright (c) 2015年 msyk.net. All rights reserved.
//

import UIKit

class DataViewController: UIViewController {

    @IBOutlet weak var dataLabel: UILabel!
    var dataObject: AnyObject?
    @IBOutlet weak var photoView: UIImageView?

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

override func viewWillAppear(animated: Bool) {
    super.viewWillAppear(animated)
    if let obj: AnyObject = dataObject {
        self.dataLabel!.text = obj.description
        let imagePath = NSBundle.mainBundle()
            .pathForResource(obj.description, ofType: "jpg")
        photoView?.image = UIImage(contentsOfFile: imagePath!)
        photoView?.contentMode = .ScaleAspectFit
    } else {
        self.dataLabel!.text = ""
    }
}
}

